ray-operator/test/sampleyaml/rayservice_test.go

```diff
@@ -43,18 +43,19 @@ func TestRayService(t *testing.T) {
 			g.Expect(err).NotTo(HaveOccurred())
 			g.Expect(rayService).NotTo(BeNil())
 
-			var rayClusterName string
+			test.T().Logf("Waiting for RayService %s/%s to running", rayService.Namespace, rayService.Name)
+			g.Eventually(RayService(test, rayService.Namespace, rayService.Name), TestTimeoutMedium).
+				Should(WithTransform(RayServiceStatus, Equal(rayv1.Running)))
+			// Get the latest rayService
+			rayService, err = GetRayService(test, namespace.Name, rayServiceFromYaml.Name)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(rayService).NotTo(BeNil())
+
 			// Wait for RayCluster name to be populated
-			g.Eventually(func(g Gomega) {
-				rs, err := GetRayService(test, namespace.Name, rayServiceFromYaml.Name)
-				g.Expect(err).NotTo(HaveOccurred())
-				if rs.Status.PendingServiceStatus.RayClusterName != "" {
-					rayClusterName = rs.Status.PendingServiceStatus.RayClusterName
-				} else if rs.Status.ActiveServiceStatus.RayClusterName != "" {
-					rayClusterName = rs.Status.ActiveServiceStatus.RayClusterName
-				}
-				g.Expect(rayClusterName).NotTo(BeEmpty())
-			}, TestTimeoutShort).Should(Succeed())
+			g.Eventually(RayService(test, namespace.Name, rayServiceFromYaml.Name), TestTimeoutShort).Should(
+				WithTransform(UnderlyingRayClusterName, Not(BeEmpty())),
+			)
+			rayClusterName := UnderlyingRayClusterName(rayService)
 
 			test.T().Logf("Waiting for RayCluster %s/%s to be ready", namespace.Name, rayClusterName)
```

Comment:
It is possible for a healthy RayService to have its underlying RayCluster status not be `rayv1.Ready`, and for some worker Pods to be not ready. We can remove the checks of `Equal(rayv1.Ready)` and `WorkerPods(test, rayCluster)`.

Reply:
Yeah, you are correct. We only need the `Head Pod` to be ready.